pipeline/scripts/filter_transcripts.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+'''
+###########################################################################
+#
+# This file is part of Cpipe.
+#
+# Cpipe is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, under version 3 of the License, subject
+# to additional terms compatible with the GNU General Public License version 3,
+# specified in the LICENSE file that is part of the Cpipe distribution.
+#
+# Cpipe is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Cpipe.    If not, see <http:#www.gnu.org/licenses/>.
+#
+###########################################################################
+# Purpose:
+#   filter tab separated output based on the transcript:
+#   remove XMs that have an NM at the same location
+# Usage:
+#   filter_transcripts.py < variants.tsv 1>filtered.tsv
+####################################################################################
+'''
+
+import argparse
+import collections
+import datetime
+import sys
+
+TRANSCRIPT_COLUMN = 'Feature'
+IDENTIFYING_COLUMNS = ['CHROM', 'POS', 'REF', 'ALT']
+
+def filter_tsv(instream, outstream, log):
+    '''
+        read instream and filter out XM transcripts that are in a location that already has an NM transcript
+    '''
+    # get header
```

Comment:
I'll do this in the next iteration. It's slightly awkward using csvreader/writer because I want to be certain that when I write the record back out I write the exact unmodified line.